### PR TITLE
[Fixes  #7269] Reduce information returned to users to only what is strictly required and accessible

### LINKED ISSUE #7269: Users information leak
## Expected Behavior

Only administrators must be able to retrieve a full list of users including their personal information.

## Actual Behavior

GeoNode's /owners API endpoint returns the full list of users, including their personal information (!!!!).

## Steps to Reproduce the Problem

  1. Go to layers page
  2. On inspect, In the network tab, Check response for owners request. The list returned includes all attributes on a user.
  3. Also check on the user's profile, more information is provided.

## Specifications

  - GeoNode version: master
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker):
  - Platform:
  - Additional details:

### PR DESCRIPTION
**Pull request by marthamareal**
<Include a few sentences describing the overall goals for this Pull Request>
Protects user's information by returning limited attributes on their profile to anonymous and non-admin users.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Review by afabiani — CHANGES_REQUESTED** [review]
When connecting with a standard user, trying to fetch the list of people, I see this

image

**Comment by marthamareal** [comment]
> When connecting with a standard user, trying to fetch the list of people, I see this
> 
> image

@afabiani  This is because only admins can access all information on other users, For logged-in and anonymous users, they only have access to (usernames) for other users.

**Comment by giohappy** [comment]
@marthamareal isn't the username enough in that template? Why is "afabiani" showing up there and not the other two usernames?

**Comment by marthamareal** [comment]
> @marthamareal isn't the username enough in that template? Why is "afabiani" showing up there and not the other two usernames?

Because the template is checking for empty first_name [here](https://github.com/GeoNode/geonode/blob/master/geonode/people/templates/people/_profile_list_item.html#L12).

**Comment by giohappy** [comment]
@marthamareal I think it's fine to also return first name and last name. We can also return the avatar if the template requires it, but we could remove it from the API response, can't we?

**Comment by marthamareal** [comment]
The template uses the data that is already modified from the api.

**Comment by afabiani** [comment]
@marthamareal we can probably remove the whole card if the username is not present?

@giohappy I would probably add an option to the users in order to allow them to make the profile public or not. This is because IMHO most of the use cases need the people list to be visibile publicly (or at least most of it).

**Comment by t-book** [comment]
>  I would probably add an option to the users in order to allow them to make the profile public or not.

👍

**Comment by giohappy** [comment]
> @giohappy I would probably add an option to the users in order to allow them to make the profile public or not. This is because IMHO most of the use cases need the people list to be visibile publicly (or at least most of it).

In general clients ask for a "private by default" policy regarding resources and users information. I'm fine with adding an option to make one's profile public, but I would do it in a separate issue.

We have to take a decision on this. What we are implementing here is:

1 - return restricted users info to non-admin users
2 - only return users which have published at least a resource which is visible to the current user

If we add the option to let the user to make his profile public we should override both these restrictions, i.e. show the public profile and independently from having published a resource or not. Is this what you expect @afabiani and @t-book?

**Comment by giohappy** [comment]
Ok, we can move this topic to a next phase and keep returning all the users. However I have a question: what's the difference between `owners/` and `profiles/` endpoints, except some different field in the response?

**Comment by afabiani** [comment]
The owners should return the facets for the counters on the resources detail (if I recall this correctly)

**Comment by giohappy** [comment]
> The owners should return the facets for the counters on the resources detail (if I recall this correctly)

profiles/ details return the facets.
owners/ detaild seems broken, it always returns null for count.

**Comment by giohappy** [comment]
@marthamareal this topic requires some more discussion. Moreover this API will be replaced by the newer API v2, so I wouldn't spend more time on this.

Please remove the counts==0 check, so we revert to returning all the users (with reduced information)

File: geonode/api/api.py [hunk 1]
```diff
@@ -126,7 +127,14 @@ def to_json(self, data, options=None):
         counts = self.get_resources_counts(options)
         if 'objects' in data:
             for item in data['objects']:
-                item['count'] = counts.get(item['id'], 0)
+                count = counts.get(item['id'], 0)
+                if count == 0 and 'owners' in item.get('resource_uri', ''):
```
Comment (marthamareal) [inline geonode/api/api.py:131]: @giohappy I had provided this check to consider the count of resources on only owners resource, what do you think?

Comment (giohappy) [reply geonode/api/api.py:131]: To be honest I don't like that hardcoded "owners" check, especially inside a generic method. Let's revert to eh previous behaviour. This PR will only reduce the returned fields but we'll keep returning users as before. We will review this with new API v2

**Comment by marthamareal** [comment]
I have made only (`avatar`, `first_name`, `last_name`, and `username`) public fields on getting owners and this also applies when getting profiles.

**Comment by giohappy** [comment]
what about the organization @marthamareal? Isn't it needed by the template?

**Review by afabiani — APPROVED** [review]

**Review by giohappy — APPROVED** [review]

**Pull request merged**